src/read.rs

```diff
@@ -30,6 +33,24 @@ impl<'a> BufReadDecoderError<'a> {
     }
 }
 
+impl<'a> fmt::Display for BufReadDecoderError<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+      match *self {
+          BufReadDecoderError::InvalidByteSequence(_) => write!(f, "invalid byte sequence"),
```

Comment:
consider also including what the bad bytes are:
```suggestion
          BufReadDecoderError::InvalidByteSequence(bytes) => write!(f, "invalid byte sequence {:x?}", bytes),
```

Reply:
Good call. I also added some padding and reformatted a tiny bit, to respect rustfmt's default line length preferences.